### PR TITLE
Constants

### PR DESCRIPTION
**Pull request by Arthur-Milchior**
## Purpose / Description
Two related purpose
Having the code more readable, by using named constants
Being more similar to anki code, by using the same constant name

Note that it is almost certain that I missed some numbers which should have been replaced by constants. It's almost impossible to check for every single 0 and see whether it represents any number which we consider in this PR.

## How Has This Been Tested?
I went through all commits, and ensured that in each commit, a single number is removed, and it is replaced by a single constant. So that no semantic change occur in any commit

## Learning (optional, can help others)
_Describe the research stage_
I should really have done a change where each commit corresponds to a single number the first time. It's so much faster to proof-read than doing a commit for the whole set of changes.

_Links to blog posts, patterns, libraries or addons used to solve this problem_

## Checklist
_Please, go through these checks before submitting the PR._

- [x] You have not changed whitespace unnecessarily (it makes diffs hard to read)
- [x] You have a descriptive commit message with a short title (first line, max 50 chars).
- [x] Your code follows the style of the project (e.g. never omit braces in `if` statements) 
- [ ] You have commented your code, particularly in hard-to-understand areas
- [x] You have performed a self-review of your own code

### CODE REVIEW

**Comment by Arthur-Milchior** [comment]
I may push more changes on top of all of this if I find other places where I should have used constants

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java [hunk 1]
```diff
@@ -1375,13 +1376,15 @@ public static void updateSearchItemQA(Context context, Map<String, String> item,
...
         item.put("edited", LanguageUtil.getShortDateFormatFromS(c.note().getMod()));
         // interval
-        int type = c.getType();
-        if (type == 0) {
-            item.put("interval", context.getString(R.string.card_browser_interval_new_card));
-        } else if (type == 1) {
-            item.put("interval", context.getString(R.string.card_browser_interval_learning_card));
-        } else {
-            item.put("interval", Utils.timeSpan(context, c.getIvl()*86400));
+        switch (c.getType()) {
```
Comment (timrae) [inline AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java]: Codacy Issue found: [A switch statement does not contain a break](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Issues
======
- Added 1
           

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  2
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 1 above)
Comment (mikehardy) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:1379]: I understand for changes like this PR you touch code far and wide, and actually *cleaning* it all up (vs mechanically changing it) has negative value. This should be left alone - resolved, from my perspective

**Comment by mikehardy** [comment]
Hey @Arthur-Milchior  :-) - This is basically a copy of what went in Anki Desktop yes? I remember scanning through it and discussing - just checking

I'll mark it "waiting for feedback" - just let me know when you feel comfortable with it and I'll scan through and merge it

Thanks!

**Comment by mikehardy** [comment]
Some sort of compile error right now though - definitely needs a clean `./gradlew assembleDebug` run at least prior to final commits:

```
/home/travis/build/ankidroid/Anki-Android/AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/ContentProviderTest.java:801: error: cannot find symbol

        assertEquals("card is initial new", Consts.CARD_TYPE_NEW, card.getQueue());

                                            ^

  symbol:   variable Consts

  location: class ContentProviderTest

<============-> 96% EXECUTING [1m 0s]> :AnkiDroid:compileDebugAndroidTestJavaWithJavac> :AnkiDroid:mergeDexDebuge/travis/build/ankidroid/Anki-Android/AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/ContentProviderTest.java:847: error: cannot find symbol

        Assert.assertNotEquals("Card is not user-buried before test", Consts.QUEUE_TYPE_SIBLING_BURIED, card.getQueue());

                                                                      ^

  symbol:   variable Consts

  location: class ContentProviderTest

/home/travis/build/ankidroid/Anki-Android/AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/ContentProviderTest.java:872: error: cannot find symbol

        assertEquals("Card is user-buried", Consts.QUEUE_TYPE_SIBLING_BURIED, cardAfterUpdate.getQueue());

                                            ^

  symbol:   variable Consts

  location: class ContentProviderTest

/home/travis/build/ankidroid/Anki-Android/AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/ContentProviderTest.java:895: error: cannot find symbol

        Assert.assertNotEquals("Card is not suspended before test", Consts.QUEUE_TYPE_SUSPENDED, card.getQueue());

                                                                    ^

  symbol:   variable Consts

  location: class ContentProviderTest

/home/travis/build/ankidroid/Anki-Android/AnkiDroid/src/androidTest/java/com.ichi2.anki.tests/ContentProviderTest.java:920: error: cannot find symbol

        assertEquals("Card is suspended", Consts.QUEUE_TYPE_SUSPENDED, cardAfterUpdate.getQueue());

                                          ^

  symbol:   variable Consts

  location: class ContentProviderTest

5 errors

```

**Comment by Arthur-Milchior** [comment]
I still need to figure out how to start test from android studio. Still is a mystery to me.
The variable names are a copy from anki's code. However, the change themselves varies, the code being different as soon as you leave libanki. 
In particular, one commit replace a variable by a switch/case. This is not in original anki, since switch/case does not exists in python

File: AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java (hunk 1 above)
Comment (Arthur-Milchior) [reply AnkiDroid/src/main/java/com/ichi2/anki/CardBrowser.java:1379]: I don't undertsand your message at all. Do you mean that the last commit has negative value for ankidroid ? Or only for the software which does quality testing ?
When you say "left alone" do you mean that this commit is left in the PR ? Or that the original code is left as it is until some "mechanical cleaning" takes care of it (no idea what would solve this kind of error)
Anyway, if the problem is only that I didn't put a break at the end of the default case, I can always add it

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  2
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

**Comment by Arthur-Milchior** [comment]
The problem I currently hvae is that, even if I try to run the test from master, `testQueryAllModels` in `ContentProviderTest` fails.
I tried clicking on "building" and sending this version to my phone, in order to see if it helps. I thought maybe running the test does not recompile everything (I doubt it is the case, since it takes time to build each time I change the current branch), so I don't really know how to succesfully resolve this bug if I still have bug on master (I would have expected to do a dichotomy to see where the bug is introduced. Since checking all changes one by one does not show me anything wrong)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  2
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  2
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  2
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  2
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

**Comment by mikehardy** [comment]
I don't use Android Studio to run tests, I use the command line in the top level directory
The things I normally run are:
1. A full compile check: `./gradlew clean assembleDebug` 
1. A non-emulator test run: `./gradlew clean jacocoUnitTestReport`
1. A full (with emulator or test device) test run: `./gradlew clean jacocoTestReport`

The full compile check would surface this problem, and even if you have issues running the emulators or similar, the second task - a non-emulator test run - should work for everyone

An analog within Android Studio for a full compile run would be to clean and build the project, otherwise Android Studio just checks the files you touch, and doesn't do the full compile if I understand correctly

Separately - my comment on fixing the Codacy warning having negative value is that with really broad changes like this I prefer to keep the change as focused as possible, even ignoring Codacy warnings. To fix all the Codacy quality flags at once would make the diff quite unwieldy, which has negative value for me. The change you propose in general has positive value, I'm excited to merge it once it's in shape!

Cheers

**Comment by timrae** [comment]
Codacy Here is an overview of what got changed by this pull request:


```diff

Clones added
============
- AnkiDroid/src/main/java/com/ichi2/libanki/SchedV2.java  1
- AnkiDroid/src/main/java/com/ichi2/libanki/Sched.java  1
         
```


See the complete overview on [Codacy](https://app.codacy.com/manual/timrae/Anki-Android/pullRequest?prid=5281675&bid=17291795)

**Comment by Arthur-Milchior** [comment]
The problem starts to appear on 
https://github.com/ankidroid/Anki-Android/pull/5888/commits/aa6f2d0b92d0bf51393992dcbcbc89a6152ced4c
`./gradlew clean assembleDebug` and `./gradlew clean jacocoUnitTestReport`
but `./gradlew clean jacocoTestReport` (with a device connected) fails in a strange way.

> 
> > Configure project :AnkiDroid
> WARNING: The following project options are deprecated and have been removed: 
> android.enableUnitTestBinaryResources
> The raw resource for unit test functionality is removed.
> 
> 
> Signing not ready for Gradle Play Publisher. Be sure to specify a signingConfig for variant 'release'.
> 
> > Configure project :api
> WARNING: The following project options are deprecated and have been removed: 
> android.enableUnitTestBinaryResources
> The raw resource for unit test functionality is removed.
> 
> 
> 
> > Task :AnkiDroid:stripDebugDebugSymbols FROM-CACHE
> WARNING: Compatible side by side NDK version was not found for android.ndkVersion '21.0.6113669'
> Compatible side by side NDK version was not found for android.ndkVersion '21.0.6113669'
> 
> > Task :AnkiDroid:connectedDebugAndroidTest
> Starting 0 tests on SM-A505U1 - 9
> 
> com.android.builder.testing.ConnectedDevice > No tests found.[SM-A505U1 - 9] FAILED 
> No tests found. This usually means that your test classes are not in the form that your test runner expects (e.g. don't inherit from TestCase or lack @Test annotations).
> 
> > Task :AnkiDroid:connectedDebugAndroidTest FAILED
> 
> FAILURE: Build failed with an exception.
> 
> * What went wrong:
> Execution failed for task ':AnkiDroid:connectedDebugAndroidTest'.
> > There were failing tests. See the report at: file:///home/arthur/anki/Anki-Android/AnkiDroid/build/reports/androidTests/connected/index.html
> 
> * Try:
> Run with --stacktrace option to get the stack trace. Run with --info or --debug option to get more log output. Run with --scan to get full insights.
> 
> * Get more help at https://help.gradle.org
> 
> Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
> Use '--warning-mode all' to show the individual deprecation warnings.
> See https://docs.gradle.org/6.2.1/userguide/command_line_interface.html#sec:command_line_warnings
> 
> BUILD FAILED in 30s
> 71 actionable tasks: 13 executed, 58 from cache
> 


In android studio it does 
2020-04-01-164722_1813x344_scrot

On the other hand https://github.com/ankidroid/Anki-Android/pull/5888/commits/adb56e3242193b405e3cb8896281097e20642f6b works perfectly
And I did re read a lot of time the commits which breaks test and have no idea what is occurring

**Comment by mikehardy** [comment]
the connected tests (`./gradlew jacocoTestReport`) in Android Studio is not reliable. It fails in the way you notice for me as well. For that reason I never run it from Android Studio. But I have not seen it fail from the command line, that's definitely odd. At the same time, a `./gradlew clean assembleDebug` and or adding `jacocoUnitTestReport` should be adequate to surface compile issues and get this PR in shape

**Comment by Arthur-Milchior** [comment]
I want on the head of the branc constants. I applied the three commands you gave. All three works sucessfuly.

**Comment by Arthur-Milchior** [comment]
Okay, it seems to work now. I've no idea why... strange.

**Comment by Arthur-Milchior** [comment]
To be clear, I consider that this PR can be merged. As I said, I may well have forgotten some numbers, given the size of the codebase. But this can always be corrected later when those numbers are found. Right now, I don't intend to search them more

**Comment by mikehardy** [comment]
Got it - let me scan through and merge, then it may conflict again with #5550 so be patient if that happens. Wide-ranging refactors like this are useful but they are difficult to get in because they touch so many things.

**Comment by Arthur-Milchior** [comment]
A lot of things already conflict with #5550, so I don't really worry for this specific purpose.
Thanks for merging. I believe it'll make editing the code far easier now

**Comment by Arthur-Milchior** [comment]
May I please ask you, if possible, to merge the PR and not to put all of it into a single commit. Because now, the new commit is really hard to read and check. And that's going to be a problem if there is a bug to try to understand and remove it. (And there is already a bug I found)

**Comment by mikehardy** [comment]
Sorry about the squash+merge vs rebase, I normally try to balance commit history (which I read a ton as release manager) with diffs as I choose between squash merge or rebase merge. Maybe got the balance wrong on this one

**Pull request merged**